### ISSUE TITLE
Cocos assert failed

### ISSUE THREAD

**Issue by apocarteres** [TopSecretGames/game#20]
After recent checkout i noticed that Cocos complains about kinda inconsistency when i run tests:

<pre>
cocos2d: fullPathForFilename: No file found at data/map1.tmx. Possible missing file.
Assert failed: Argument must be non-nil
Assertion failed: (child != nullptr), function addChild, file /Users/sandro/work/repo/topsecretgames/game/src/cocos2d/cocos/2d/CCNode.cpp, line 981.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
run-tests is a Catch v1.2.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
Tests that game event listeners got notified well
-------------------------------------------------------------------------------
GameControllerTest.cpp:12
...............................................................................

GameControllerTest.cpp:12: FAILED:
due to a fatal error condition:
  SIGABRT - Abort (abnormal termination) signal

===============================================================================
test cases: 1 | 1 failed
assertions: 1 | 1 failed
</pre>

**Comment by apocarteres** [comment]
Seems that <code>GameController.cpp:Init()</code> crashes on trying to load map from <code>data/map1.tmx</code> which doesn't exist on test scope.

To correct fix it we need solve #23 first

**Comment by stavenko** [comment]
This bug could be solved without DI issue.
